pkg/apis/gcpprovider/v1beta1/gcpmachineproviderconfig_types.go

```diff
@@ -75,3 +76,24 @@ type GCPServiceAccount struct {
 	Email  string   `json:"email"`
 	Scopes []string `json:"scopes"`
 }
+
+// GCPEncryptionKeyReference describes the encryptionKey to use for a disk's encryption.
+type GCPEncryptionKeyReference struct {
```

Comment:
Do we need this extra wrapper struct?  It might be useful if there are other types of keys we might need to support in the future, but I don't know if those exist?

Reply:
You've hit the nail on the head 😅  You can also specify other options normally (such as providing a raw key) but I didn't want to implement those in a first pass. To make sure we can support those in a future I added the wrapper struct

For reference [this](https://godoc.org/google.golang.org/api/compute/v1#CustomerEncryptionKey) is the struct on the compute module, extras there are ServiceAccount and RawKey. I'm trying to test this now and we might actually need the ServiceAccount anyway 🤔